test/ctia/logger_test.clj

```diff
@@ -0,0 +1,36 @@
+(ns ctia.logger-test
+  (:require [ctia.logging :refer :all]
+            [ctia.events :as c :refer :all]
+            [clojure.test :as t :refer :all]
+            [schema.test :as st]
+            [clojure.tools.logging :as log]
+            [clojure.string :as str]))
+
+(use-fixtures :once st/validate-schemas)
+
+(deftest test-setup
+  (init!)
+  (let [{b :chan-buf c :chan m :mult :as ev} @central-channel]
+    (log-channel ev)
+    (send-create-event "tester" {} "TestModelType" {:data 1})
+    (Thread/sleep 100))) ;; wait until the go loop is done
+  
+(deftest test-logged
+  (init!)
+  (let [{b :chan-buf c :chan m :mult :as ev} @central-channel
+        sb (StringBuilder.)
+        ll log/log*
+        patched-log (fn [logger level throwable message]
+                      (.append sb message)
+                      (.append sb "\n"))]
+    (alter-var-root (var log/log*) (constantly patched-log))
```

Comment:
It is indeed a code smell. I nearly wrote a macro to duplicate the functionality of with-redefs, since I thought it would be handy to have such a thing. Thank you for this!

The correct way to do it would be to register a LogWriter that picks up the logs and presents them back. Unfortunately, SLF4J wrapped and hid this functionality (which I learnt from the sources on Apache after spending some time thinking I could use a writer like I just said). Monkey patching the log* is a hack to get around this.